tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageSpecificCheckResolver.cs

```diff
@@ -0,0 +1,155 @@
+
+        if (!Directory.Exists(packagePath))
+        {
+            throw new ArgumentException($"Package path does not exist: {packagePath}", nameof(packagePath));
+        }
+
+        _logger.LogDebug("Resolving language-specific check for package at {PackagePath}", packagePath);
+
+        // Use repository detection service to identify the language
+        var detectedLanguage = await DetectLanguageAsync(packagePath);
+        
+        if (string.IsNullOrEmpty(detectedLanguage))
+        {
+            _logger.LogWarning("No language detected for package at {PackagePath}", packagePath);
+            return null;
+        }
+
+        // Find the appropriate language-specific check implementation
+        var specificLanguageCheck = _languageChecks.FirstOrDefault(check => 
+            string.Equals(check.SupportedLanguage, detectedLanguage, StringComparison.OrdinalIgnoreCase));
+        
+        if (specificLanguageCheck != null)
+        {
+            _logger.LogInformation("Selected {Language} check service for package at {PackagePath}", 
+                specificLanguageCheck.SupportedLanguage, packagePath);
+            return specificLanguageCheck;
+        }
+
+        _logger.LogWarning("No language-specific check service found for detected language '{Language}' at package path {PackagePath}", 
+            detectedLanguage, packagePath);
+        return null;
+    }
+
+    private async Task<string?> DetectLanguageAsync(string packagePath)
+    {
+        try
+        {
+            var repositoryPath = _gitHelper.DiscoverRepoRoot(packagePath);
+            if (string.IsNullOrEmpty(repositoryPath))
+            {
+                return null;
+            }
+
+            // Read Language-Settings.ps1 file to determine language
+            var languageSettingsPath = Path.Combine(repositoryPath, "eng", "scripts", "Language-Settings.ps1");
+            if (!File.Exists(languageSettingsPath))
+            {
+                _logger.LogWarning("Language-Settings.ps1 not found at {LanguageSettingsPath}", languageSettingsPath);
+                return null;
+            }
+
+            var language = await ExtractLanguageFromFileAsync(languageSettingsPath);
+            if (!string.IsNullOrEmpty(language))
+            {
+                _logger.LogDebug("Detected language: {Language} from Language-Settings.ps1 at {Path}", language, languageSettingsPath);
+                return language;
+            }
+
+            return null;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error detecting language for package path: {PackagePath}", packagePath);
+            return null;
+        }
+    }
+
+    private async Task<string?> ExtractLanguageFromFileAsync(string filePath)
+    {
+        try
+        {
+            var options = new PowershellOptions([". '" + filePath.Replace("'", "''") + "'; Write-Output $Language"], 
+                workingDirectory: Path.GetDirectoryName(filePath));
```

Comment:
I am going full circle back to to this because imo this is the most reliable way to get the language and we are already calling pwsh scripts elsewhere